exotica_core/src/kinematic_tree.cpp

```diff
@@ -995,6 +997,63 @@ void KinematicTree::SetJointLimitsUpper(Eigen::VectorXdRefConst upper_in)
     UpdateJointLimits();
 }
 
+void KinematicTree::SetJointLimitsVelocity(Eigen::VectorXdRefConst velocity_in)
+{
+    if (velocity_in.rows() == num_controlled_joints_)
+    {
+        for (unsigned int i = 0; i < num_controlled_joints_; ++i)
+        {
+            controlled_joints_[i].lock()->velocity_limit[0] = velocity_in(i);
+        }
+    }
+    else
+    {
+        ThrowPretty("Got " << velocity_in.rows() << " but " << num_controlled_joints_ << " expected.");
+    }
+
+    UpdateJointLimits();
+}
+
+void KinematicTree::SetJointLimitsAcceleration(Eigen::VectorXdRefConst acceleration_in)
+{
+    if (acceleration_in.rows() == num_controlled_joints_)
+    {
+        for (unsigned int i = 0; i < num_controlled_joints_; ++i)
+        {
+            controlled_joints_[i].lock()->acceleration_limit[0] = acceleration_in(i);
+        }
+
+        has_acceleration_limit = true;
+    }
+    else
+    {
+        ThrowPretty("Got " << acceleration_in.rows() << " but " << num_controlled_joints_ << " expected.");
+    }
+
+    UpdateJointLimits();
+}
+
+void KinematicTree::SetFloatingBaseLimitsPosXYZEulerZYX(
+    const std::vector<double>& lower, const std::vector<double>& upper, const std::vector<double>& velocity, const std::vector<double>& acceleration)
+{
+    if (controlled_base_type_ != BaseType::FLOATING)
+    {
+        ThrowPretty("This is not a floating joint!");
+    }
+    if (lower.size() != 6 || upper.size() != 6)
```

Comment:
You need to also check the dimension of velocity and acceleration

Reply:
And should their sizes also be 6? Not familiar with what this function does intuitively so not sure

Reply:
ie: is this OK?

```if (lower.size() != 6 || upper.size() != 6 || velocity.size() != 6 || acceleration.size() != 6)
    {
        ThrowPretty("Wrong limit data size!");
    }```

Reply:
And the same for SetPlanarBaseLimitsPosXYEulerZ (line 1079)? But size=3:

```
    if (lower.size() != 3 || upper.size() != 3 || velocity.size() != 3 || acceleration.size() != 3)
    {
        ThrowPretty("Wrong limit data size!");
    }
```

Reply:
Vlad confirmed that this sizes should be 6 and 3 respectively